### PR TITLE
Download responses check dialog

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

part of https://github.com/cds-snc/platform-forms-client/issues/1978 (see "Out of scope" below)

Adds check for responses that haven't been dowloaded prior to allow delete.  See: https://github.com/cds-snc/platform-forms-client/pull/1998

- Adds API call + new screen when responses still exist
- Adds Cypress component tests
- Adds error handling for failed API call

**EN**
Screen Shot 2023-04-28 at 11 29 31 AM

**FR**
Screen Shot 2023-04-28 at 11 30 12 AM

# Test instructions | Instructions pour tester la modification

- Create a form
- Adds some responses
- Try to delete the form

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

- This PR does not include updates for Failed to delete based on responses still exists.  That can be done in a follow-up PR.

https://github.com/cds-snc/platform-forms-client/issues/1978#issuecomment-1525746948

**Started PR to update Toast Message** 
https://github.com/cds-snc/platform-forms-client/pull/2009

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [ ] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: public/static/locales/en/form-builder.json [hunk 1]
```diff
@@ -291,7 +291,16 @@
   "formatItalic": "Format italic",
   "formatNumberedList": "Format numbered list",
   "formConfirmationMessage": "Form confirmation message",
+  "formDeleteResponses": {
+    "title": "Get responses before deleting form",
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:295]:
```suggestion
    "title": "Confirm responses before deleting form",
```

File: public/static/locales/en/form-builder.json [hunk 2]
```diff
@@ -291,7 +291,16 @@
...
+    "message1": "This form has been published and is receiving responses. You have responses to download and confirm.",
+    "message2": "Before deleting this form:",
+    "message3": "1. Download all responses.",
```
Comment (anikbrazeau) [inline public/static/locales/en/form-builder.json:298]:
```suggestion
    "message3": "1. Download all new responses.",
```

File: public/static/locales/fr/form-builder.json [hunk 3]
```diff
@@ -291,7 +291,16 @@
   "formatItalic": "Mettre en italique",
   "formatNumberedList": "Mettre en liste numérotée",
   "formConfirmationMessage": "Message de confirmation du formulaire",
+  "formDeleteResponses": {
+    "title": "Retirer les réponses avant de supprimer le formulaire",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:295]:
```suggestion
    "title": "Confirmer les réponses avant de supprimer le formulaire",
```

File: public/static/locales/fr/form-builder.json [hunk 4]
```diff
@@ -291,7 +291,16 @@
...
+    "message1": "Ce formulaire est publié et reçoit des réponses. Vous avez des réponses à télécharger et à confirmer.",
+    "message2": "Avant de supprimer ce formulaire :",
+    "message3": "1. Télécharger toutes les réponses.",
```
Comment (anikbrazeau) [inline public/static/locales/fr/form-builder.json:298]:
```suggestion
    "message3": "1. Télécharger toutes les nouvelles réponses.",
```

File: components/form-builder/app/shared/ConfirmFormDeleteDialog.tsx [hunk 5]
```diff
@@ -28,18 +81,55 @@ export const ConfirmFormDeleteDialog = ({
...
+      </Dialog>
+    );
+  }
+
   return (
     <Dialog handleClose={handleClose} dialogRef={dialog} actions={actions}>
       <div className="p-5">
         <div className="px-10 flex justify-center">
           <Image
-            layout="fixed"
```
Comment (timarney) [inline components/form-builder/app/shared/ConfirmFormDeleteDialog.tsx:42]: Next 13 was showing a warning here

**Review by thiessenp-cds — COMMENTED** [review]
Nice work! 
I gave this a quick try locally and get a dialog with the unhandled error (something went wrong). From what I can see the client request to the api is correct and the response sent back is a 400 bad request. Odd
Though if this works for you locally it could just be my setup. Other than this the PR LGTM :)
Screenshot 2023-04-28 at 2 48 42 PM

**Comment by craigzour** [comment]
> Nice work! I gave this a quick try locally and get a dialog with the unhandled error (something went wrong). From what I can see the client request to the api is correct and the response sent back is a 400 bad request. Odd Though if this works for you locally it could just be my setup. Other than this the PR LGTM :) Screenshot 2023-04-28 at 2 48 42 PM

I have the same issue. Trying to investigate as it should not behave this way.

**Comment by timarney** [comment]
@thiessenp-cds @craigzour 

I pushed some updates I was looking for the wrong status / response

It should be 200 + 

```
{ numberOfUnprocessedSubmissions: <number> }
```

**Comment by timarney** [comment]
@craigzour should we check to see if the form ID was using "the vault" pre API call or is that handled?

i.e if they are using email does that cause issues?

**Review by craigzour — APPROVED** [review]
LGTM :) Nice work!

**Comment by craigzour** [comment]
> @craigzour should we check to see if the form ID was using "the vault" pre API call or is that handled?
> 
> i.e if they are using email does that cause issues?

It is handled by the function that fetches all submissions as it will not find any if the user has selected email instead of vault.

**Pull request merged**